plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/MavenFacet.java

```diff
@@ -49,7 +49,7 @@
   @Nullable
   Content get(MavenPath path) throws IOException;
 
-  void put(MavenPath path, Payload payload) throws IOException;
+  Content put(MavenPath path, Payload payload) throws IOException;
```

Comment:
Good to see it's actually being used now :p

BTW we should make sure that it's not too expensive to construct the content. Otherwise we're optimizing one facet which uses the return value (proxy) at the expense of another that doesn't (hosted). As long as it's just marshalling existing data into `Content` then it should be ok.